flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolImplTest.java

```diff
@@ -738,6 +740,95 @@ public void testCalculationOfTaskExecutorUtilization() throws Exception {
 		}
 	}
 
+	@Test
+	public void testOrphanedAllocationCanBeRemapped() throws Exception {
+		try (SlotPoolImpl slotPool = createSlotPoolImpl()) {
+			final List<AllocationID> allocationIds = new ArrayList<>();
+			resourceManagerGateway.setRequestSlotConsumer(
+				slotRequest -> allocationIds.add(slotRequest.getAllocationId()));
+
+			final List<AllocationID> canceledAllocations = new ArrayList<>();
+			resourceManagerGateway.setCancelSlotConsumer(canceledAllocations::add);
+
+			setupSlotPool(slotPool, resourceManagerGateway, mainThreadExecutor);
+			final Scheduler scheduler = setupScheduler(slotPool, mainThreadExecutor);
```

Comment:
why do we use `Scheduler` to unit test `SlotPoolImpl`?
why not to call `SlotPoolImpl` directly, like in `testFailingAllocationFailsRemappedPendingSlotRequests`?

Reply:
You are right. I have dropped scheduler in the newly added cases. 
However, this unnecessary complication is a common problem of most of the SlotPool tests. I think we can simplify them in a separate task as well.